### DISCUSSION TITLE
Why doesn't Writer\Xls\save return anything, but Root\save does?

### DISCUSSION THREAD

**Discussion by MikeWillis** [PHPOffice/PhpSpreadsheet#3676] [Q&A] [answered]
The documentation for saving an xls file states that it returns `void` (https://phpoffice.github.io/PhpSpreadsheet/classes/PhpOffice-PhpSpreadsheet-Writer-Xls.html#method_save)

The xls save method calls the root save method(Shared\OLD\PPS\ROOT\save) which returns `bool`, but that result is not returned by the xls save method. (https://phpoffice.github.io/PhpSpreadsheet/classes/PhpOffice-PhpSpreadsheet-Shared-OLE-PPS-Root.html#method_save).

It seems there's no way to be sure if the save was successful or not. 

Is there some way to find out?

**Comment by oleibman** [comment, marked as answer]
The `save` method for Xls, and for all other Writers, should either succeed or throw an exception. Getting to the next statement is an indication of sucess. The boolean return from `Root/save` is misleading - it will always return `true` (if it doesn't throw an exception). Adding a boolean return to Xls/save would be equally misleading.

**Reply by MikeWillis** [reply]
Thanks @oleibman, sounds like a try/catch surrounding the save should do the trick then
